src/BottomNavigation/BottomNavigation.js

```diff
@@ -0,0 +1,57 @@
+// @flow weak
+
+import React, { PropTypes, Children, cloneElement } from 'react';
+import { createStyleSheet } from 'jss-theme-reactor';
+import classNames from 'classnames';
+
+export const styleSheet = createStyleSheet('BottomNavigation', (theme) => {
+  return {
+    root: {
+      display: 'flex',
+      justifyContent: 'center',
+      height: 56,
+      backgroundColor: theme.palette.background.paper,
+    },
+  };
+});
+
+export default function BottomNavigation(props, context) {
+  const {
+    children: childrenProp,
+    className: classNameProp,
+    selectedIndex,
+    showLabel,
+    onChangeIndex,
+    ...other,
+  } = props;
+  const classes = context.styleManager.render(styleSheet);
+  const className = classNames(classes.root, classNameProp);
+
+  const children = Children.map(childrenProp, (child, index) => {
+    return cloneElement(child, {
+      selected: index === selectedIndex,
+      showLabel: child.props.showLabel !== undefined ? child.props.showLabel : showLabel,
+      onClick: onChangeIndex.bind(null, index),
```

Comment:
Yeah, that's the alternative. We would then need to create a handler on the child element. Making it using a class instead of a function. I think it worths it.